packages/salesforcedx-vscode-core/src/context/workspaceContext.ts

```diff
@@ -26,6 +29,7 @@ export class WorkspaceContext {
     const workspaceContextUtil = WorkspaceContextUtil.getInstance();
     this.onOrgChange = workspaceContextUtil.onOrgChange;
     this.onOrgChange(this.handleCliConfigChange);
+    this.onOrgChange(this.handleOrgShapeChange);
```

Comment:
registered a subscription here in the workspaceContext initialization when core starts up.